src/Neo/Network/P2P/RemoteNode.cs

```diff
@@ -234,7 +234,11 @@ internal static Props Props(NeoSystem system, LocalNode localNode, object connec
         private void SendMessage(Message message)
         {
             ack = false;
-            SendData(ByteString.FromBytes(message.ToArray(Version.AllowCompression)));
+            // Here it is possible that we dont have the Version message yet,
+            // so we need to send the message uncompressed
+            SendData(message.Command == MessageCommand.Version
+                ? ByteString.FromBytes(message.ToArray())
+                : ByteString.FromBytes(message.ToArray(Version.AllowCompression)));
```

Comment:
```suggestion
            // Here it is possible that we dont have the Version message yet,
            // so we need to send the message uncompressed
            SendData(ByteString.FromBytes(message.ToArray(Version?.AllowCompression ?? false)));
```
`ByteString.FromBytes(message.ToArray())` is telling the client that packet is compressed. Even though is not.

Removes compression flag for packet. So client doesn't try to decompress data that isn't compressed.
https://github.com/neo-project/neo/blob/befe7f64d2ef7fa8abfb02864f63bc0bcac07875/src/Neo/Network/P2P/Message.cs#L150-L152

**`ToArray()` calls `Serialize`**:
https://github.com/neo-project/neo/blob/befe7f64d2ef7fa8abfb02864f63bc0bcac07875/src/Neo/Extensions/IO/ISerializableExtensions.cs#L24-L31

Which in turn doesn't remove the compression `flag`.
https://github.com/neo-project/neo/blob/befe7f64d2ef7fa8abfb02864f63bc0bcac07875/src/Neo/Network/P2P/Message.cs#L130-L135